doc/release-notes-22350.md

```diff
@@ -0,0 +1,18 @@
+New settings
+------------
+
+- The `debug.log` file grows without bound until the node
+  is restarted, at which time it is trimmed to 10 MB. Beginning with this
+  release, you can enable log file rotation. When the `debug.log` file
+  reaches 1 MB, it's renamed to a name that includes the current date and
+  time, such as `debug-2021-06-27T21:19:42Z.log` and `debug.log` is
+  recreated as an empty file. Running `cat` or `grep` with an argument
+  `debug*.log` accesses all debug log files, including the current one
+  (`debug.log`) in time-order. The `-debuglogrotatekeep=n` configuration
+  option specifies the number of rotated debug log files that are retained
+  (default 0, log rotation disabled). The `-debugloglimit=n`
+  configuration option specifies how large the `debug.log` file must
+  become (in MB) before it is rotated (default is 10). If you enable
+  log file rotation and specify an
+  alternate debug log file name using `-debuglogfile=path`, the last
+  component of the path must be `debug.log`. (#22350)
```

Comment:
Thanks so much for your comments; these are very useful.

> ... rotation should work regardless of the filename ...

I like your idea. After giving this more thought, perhaps the best approach is to let `-debuglogfile=path` to work as usual, but create a directory with a fixed name like `rotated-logs` (or `rotated-debug-logs`) and put all the _rotated_ log files within that directory. (The active debug log file, `debug.log` or as specified by `-debuglogfile`, would be as it is today, no change.) Because the rotated log files are in a directory with a specific purpose, they can have simple names like `2021-06-27T15:10:55Z.log`. They wouldn't have to be named according to the `-debuglogfile` argument.

This would have an added benefit: As I've been running this code to test it, my data directory now looks "messy" or polluted, with all these rotated log files that usually aren't very important -- why should they be so visible and prominent?
```
$ ls ~/.bitcoin
banlist.dat   debug-2021-06-27T21:19:42Z.log  fee_estimates.dat  signet
banlist.json  debug-2021-07-09T00:38:43Z.log  indexes            testnet3
bitcoin.conf  debug-2021-07-14T16:32:30Z.log  mempool.dat        wallets
bitcoind.pid  debug-2021-07-17T14:01:27Z.log  peers.dat          wallets-empty
blocks        debug-2021-08-10T20:03:04Z.log  regtest
chainstate    debug.log                       settings.json
$
```
(And I've got only 5 rotated log files; there could be many more.) A small disadvantage is that it would be slightly more difficult to concatenate (or grep) all of the log files in time-order:
```
$ cat rotated-logs/* debug.log
```
(instead of just `cat debug*.log`) but that's not bad.

So unless there are objections, I'll make this change within the next few days.

Reply:
regarding many log files creating clutter in the root directory, i agree having a logs folder definitely makes sense. personally, i would prefer a `logs/` folder containing all the logs as this makes it clear to the user there is nothing special about the `debug.log` file vs what's in the logs directory. it also keeps things simple for concatenating and grepping. 

since the debuglogfile custom name is known, wouldn't it be possible to attempt to split the filename into a name and extension (`<name>-<timestamp>.<ext>`), and then handle the special case where the user doesn't supply an extension (i.e `logs`, which would become `logs`, `logs-<timestamp>`)?

also, feel free to wait for more feedback from others before making changes :)